app/containers/markdown/Hashtag.tsx

```diff
@@ -15,15 +21,27 @@ interface IHashtag {
 
 const Hashtag = React.memo(({ hashtag, channels, navToRoomInfo, style = [] }: IHashtag) => {
 	const { theme } = useTheme();
+	const isMasterDetail = useAppSelector(state => state.app.isMasterDetail);
+	const navigation = useNavigation<StackNavigationProp<ChatsStackParamList, 'RoomView'>>();
 
-	const handlePress = () => {
+	const handlePress = async () => {
 		const index = channels?.findIndex(channel => channel.name === hashtag);
 		if (typeof index !== 'undefined' && navToRoomInfo) {
 			const navParam = {
 				t: 'c',
 				rid: channels?.[index]._id
 			};
-			navToRoomInfo(navParam);
+			const room = await getSubscriptionByRoomId(navParam.rid!);
+			if (room) {
+				if (isMasterDetail) {
+					// Close the modal if it is open and then redirect to the channel
+					Navigation.navigate('DrawerNavigator');
+					navigation.replace('RoomView', room);
+				} else navigation.push('RoomView', room);
+			} else {
+				// Navigate to RoomInfoView if the channel is not joined
+				navToRoomInfo(navParam);
```

Comment:
Did you notice on your video the title is getting blank on this step?
Also, try to interact a bit more on this, like joining, navigating to RoomActions, etc and make sure the layout is working correctly.